src/freenet/node/useralerts/UserAlert.java

```diff
@@ -50,6 +60,18 @@ public interface UserAlert {
 	
 	public String dismissButtonText();
 	
+	public default String replyButtonText() {
+		return dismissButtonText();
+	};
+
+	public default boolean canReply() {
+		return false;
+	}
+
+	public default WeakReference<PeerNode> getSourceNode() {
+		return null;
```

Comment:
With this interface, the caller needs to deal with 2 variants of "source node not known": either the WeakReference can be `null` (when this method is not implemented), or its referent can be `null` (when cleared by the garbage collector).

Can we just return `PeerNode` here? Of course this only concerns the return type of this method; implementations that do hold a (weak)reference would just return `reference.get()`.

Reply:
That’s nicer, yes. Fixed. I also added null-safety.